range_queries/heavy_light_decomposition.cpp

```diff
@@ -0,0 +1,582 @@
+/**
+ * @file heavy_ligt_decomposition.cpp
```

Comment:
Better to remove the file name.
```suggestion
 * @file
```